network/context.go

```diff
@@ -12,8 +12,28 @@ var DialPeerTimeout = 60 * time.Second
 
 type noDialCtxKey struct{}
 type dialPeerTimeoutCtxKey struct{}
+type forceDirectDialCtxKey struct{}
```

Comment:
Why is the key a struct, not a `const string`?

Reply:
I see that it's consistent with what we've been doing so far, just wondering...

Reply:
The idea is you can't screw it up by accident. If you use a string someone else could use the string somewhere else and cause a conflict. For example, two applications could use the key word "forceDialDirect" one meaning to skip looking up the peer in some system like a DHT and the other meaning to refuse to use relays. Using an unexported struct makes this type of error basically impossible.

Reply:
That's clever. Thanks for the explanation!